EIPS/eip-5528.md

```diff
@@ -0,0 +1,183 @@
+    function escrowErrorCodeDescription(uint32 _code) external view returns (string);
+
+
+    /// @notice deposit fund(token) into escrow account.
+    /// @dev
+    ///   - seller/buyer contract should call escrow contract's function before _transfer.
+    ///   - escrow contract should update (Seller, Buyer) balance.
+    ///   - the seller can call this function to fund initial supply.
+    /// @param
+    ///   - to:
+    ///     In case of a buyer/seller contract, it must be an escrow contract address.
+    ///     In case of an escrow contract, it must be the user address that triggered this transaction.
+    ///   - _valuePayed: payable token amount
+    /// @return reason code. 0 is a success, otherwise is failure code.
+    function escrowFund(address to, uint256 amount) public returns (uint32);
+
+
+    /// @notice refund from escrow account.
+    /// @dev
+    ///   - seller/buyer contract should call escrow contract's function before _transfer.
+    ///   - escrow contract should update (Seller, Buyer) balance.
+    ///   - the seller should not call this function.
+    /// @param
+    ///   - to:
+    ///     In case of a buyer/seller contract, must be an escrow contract address.
+    ///     In case of an escrow contract, must be the user address who triggered this transaction.
+    ///   - _valuePayed: payable token amount
+    /// @return reason code. 0 is a success, otherwise is failure code.
+    function escrowRefund(address to, uint256 amount) public returns (uint32);
+
+    /// @notice withdraw token from the escrow account.
+    /// @dev
+    ///   - must be implemented in Escrow-Contract and optional for other contracts.
+    ///   - buyer is only available when escrow is successful, otherwise should call escrowRefund.
+    ///   - in case the escrow fails, the seller can refund the seller-token.
+    ///   - if the escrow is successful, the seller and buyers can get the exchanged tokens in their wallets.
+    /// @return reason code. 0 is success, otherwise is failure code.
+    function escrowWithdraw() public returns (uint32);
+
+}
+
+
+```
+
+## Rationale
+
+This standard proposes interfaces on top of the [EIP-20](./eip-20.md) standard.
+Each function should include constraint check logic.
+The escrow-contract should implement internal constraint logic such as
+ - Lock period
+ - Maximum(or minimum) number of investors
+ - Maximum(or minimum) number of tokens to fund
+ - Exchange rates of seller/buyer token
+ - KYC verification of users(It might require additional interface)
+ - etc
+
+The buyer-contract and seller-contract should not have constraint rules.
+
+Let's discuss the following functions.
+
+1. **constructor**
+
+An escrow contract will define success/failure conditions. It means constraint rules might not be changed forever (might be changed after being created for the market exchange rate.), so it guarantees escrow policy.
+
+2. **escrowFund**
```

Comment:
```suggestion
#### `escrowFund`
```
Recommended: Subheadings here make this clearer
Recommended: Use code blocks when using code.